### DISCUSSION TITLE
Setting up the SSL Certificates fot HTTPS access

### DISCUSSION THREAD

**Discussion by NudeNULL** [verdaccio/verdaccio#3473] [Q&A] [answered]
Hello
I am trying to set up the HTTPS access to Verdaccio by following the steps from this guide - https://verdaccio.org/docs/ssl

I was able to complete all of them without any errors, but the problem is that the browser is showing that the certificate is not valid, and for the first connection you have to use "proceed to the unsafe page" option in order to open the Web UI.

Is this a normal behavior, or the Certificate should be valid after these steps?
If this is normal then how could I achieve a secure connection with correct setup?

I am running the latest Verdaccio on a Windows machine.

Thanks!

**Comment by juanpicado** [comment]
Is the certificate valid ? Or did you signed yourself?

**Reply by NudeNULL** [reply]
I am pretty sure the problem is in my poor knowledge on certificates and https setup, but I just tried to do only exactly what is said in the guide I mentioned.

As I wrote, the browser is showing that the certificate is not valid when I try to access the Web UI.

I do not know the process behind signing a certificate, but from your guide I believe that I only generated the certificates, do I need to do additional steps for them to be signed and valid?

Also, the problem could be that I am not sure what is the expected result by completing that guide, maybe it is not enough to have a valid and secure connection to Verdaccio and I have to do additional steps by myself.

Thanks!

**Reply by NudeNULL** [reply]
One more thing about that guide is that I couldn't access verdaccio from outside my network until I changed the listen address from screenshot to this one https://0.0.0.0:4873/
![image](https://user-images.githubusercontent.com/15786742/200122800-a301b02c-fba1-43cf-abab-f228d2f5ca8a.png)

**Reply by NudeNULL** [reply]
I mean not exactly the address from the image, but my domain used to access verdaccio

**Comment by NudeNULL** [comment]
So to be extra clear, I just did this steps in order to set up the certificates
![image](https://user-images.githubusercontent.com/15786742/200122880-36aae820-5a19-48dc-a0af-d4bd677e9d66.png)

**Comment by NudeNULL** [comment, marked as answer]
@juanpicado, I continued to research how to secure Verdaccio and did it by using Lets Encrypt and Certbot, so I just generated the certs using the command line and certbot, and specified the paths to them in the config.yaml, restarted the instance and everything works perfect!

**Reply by juanpicado** [reply]
Awesome !!! glad you did it !! If you want to improve the docs just feel free to update this file https://github.com/verdaccio/verdaccio/blob/master/website/docs/ssl.md so others :) can benefit from your research.
